solr/core/src/java/org/apache/solr/update/processor/AtomicUpdateDocumentMerger.java

```diff
@@ -553,7 +574,15 @@ private Object getNativeFieldValue(String fieldName, Object val) {
       return val;
     }
     SchemaField sf = schema.getField(fieldName);
-    return sf.getType().toNativeType(val);
+    try {
+      return sf.getType().toNativeType(val);
+    } catch (SolrException ex) {
+      throw new SolrException(SolrException.ErrorCode.getErrorCode(ex.code()),
+          "Error converting field '" + sf.getName() + "'='" +val+"' to native type, msg=" + ex.getMessage(), ex);
```

Comment:
I don't think we want `msg` copied since it will be in the cause anyway.

Reply:
cause gets lost in the metadata section of the response so, thought this would give simpler insight into error. Also, trying to follow the same convention as other error messages in DocumentBuilder